flink-connectors/flink-connector-gcp-pubsub/src/main/java/org/apache/flink/streaming/connectors/gcp/pubsub/PubSubSource.java

```diff
@@ -264,12 +263,11 @@ private PubSubSourceBuilder(PubSubDeserializationSchema<OUT> deserializationSche
 		}
 
 		/**
-		 * Set a limit of the number of outstanding or to-be acknowledged messages.
-		 * default is 10000. Adjust this if you have high checkpoint intervals and / or run into memory issues
-		 * due to the amount of acknowledgement ids. Use {@link PubSubSource}.NO_MAX_MESSAGES_TO_ACKNOWLEDGE_LIMIT if you want to remove the limit.
+		 * Set a limit on the rate of messages per second received. This limit is per parallel instance of the source function.
+		 * Default is set to 100000 messages per second.
 		 */
-		public PubSubSourceBuilder<OUT> withMaxMessageToAcknowledge(int maxMessageToAcknowledge) {
-			this.maxMessageToAcknowledge = maxMessageToAcknowledge;
+		public PubSubSourceBuilder<OUT> withRateLimit(int messagePerSecondRateLimit) {
```

Comment:
I renamed this method to `withMessageRateLimit` in case we will have bytes rate limit in the future.